### PR TITLE
4.x clean up on node need sudo

### PR DESCRIPTION
**Pull request by liangxia**
@akostadinov @pruan-rht  
Issue described in cucushift mail list with subject "4.x clean up on nodes need sudo"

### CODE REVIEW

File: lib/host.rb [hunk 1]
```diff
@@ -524,10 +524,12 @@ def delete(file, opts={})
       end
       if opts[:raw]
         exec_raw("rm #{r} -f -- #{file}", opts)
```
Comment (pruan-rht) [inline lib/host.rb]: @liangxia why not remove the old lines?

Comment (liangxia) [reply lib/host.rb]: @pruan  We may have some situations where sudo are not configured.

Comment (pruan-rht) [reply lib/host.rb]: @liangxia if `sudo` is not configured, then this test won't work, if I read the description `clean up on node need sudo` correctly....

Comment (akostadinov) [reply lib/host.rb]: The low-level method should not assume user wants `sudo`. But the step
where deletion is done, should know. My suggestion is to fix the step(s).

Lets alter the steps necessary. Do you know where the delete step is registered as a clean-up for this failure?

Comment (liangxia) [reply lib/host.rb]: You mean this one ?
`the "/etc/origin/hostpath/<%= project.name %>" path is recursively removed on the host after scenario`

Comment (akostadinov) [reply lib/host.rb]: This one also is a little bit low level. I mean to update this step so it does remove files as admin on the host where needed. Or do you think too many places need update?

**Pull request closed without merging**